sdk/src/wallet/migration/chrysalis.rs

```diff
@@ -0,0 +1,310 @@
+    )?;
+    // write new accounts to db (with account indexation)
+    let accounts_indexation_data: Vec<u32> = new_accounts.iter().map(|account| account.index).collect();
+    stardust_db.put(
+        ACCOUNTS_INDEXATION_KEY,
+        serde_json::to_string(&accounts_indexation_data)?,
+    )?;
+    for new_account in new_accounts {
+        stardust_db.put(
+            &format!("{ACCOUNT_INDEXATION_KEY}{}", new_account.index),
+            serde_json::to_string(&new_account)?,
+        )?;
+    }
+
+    if let Some(secret_manager_dto) = secret_manager_dto {
+        // This is required for the secret manager to be loaded
+        stardust_db.put(WALLET_INDEXATION_KEY, format!("{{ \"coinType\": {IOTA_COIN_TYPE}}}"))?;
+        stardust_db.put(SECRET_MANAGER_KEY, secret_manager_dto)?;
+    }
+
+    // set db migration version
+    let migration_version = crate::wallet::migration::migrate_4::Migrate::version();
+    stardust_db.put(MIGRATION_VERSION_KEY, serde_json::to_string(&migration_version)?)?;
+
+    drop(stardust_db);
+
+    // remove old db
+    std::fs::remove_dir_all(chrysalis_storage_path)?;
+
+    Ok(())
+}
+
+pub(crate) fn migrate_from_chrysalis_data(
+    chrysalis_data: &HashMap<Vec<u8>, String>,
+    storage_path: &Path,
+    // in stronghold the keys are hashed first
+    stronghold: bool,
+) -> Result<(Vec<AccountDetailsDto>, Option<String>)> {
+    let mut new_accounts: Vec<AccountDetailsDto> = Vec::new();
+    let mut secret_manager_dto: Option<String> = None;
+
+    let account_indexation_key = if stronghold {
+        key_to_chrysalis_key(b"iota-wallet-account-indexation")
+    } else {
+        b"iota-wallet-account-indexation".to_vec()
+    };
+    if let Some(account_indexation) = chrysalis_data.get(&account_indexation_key) {
+        if let Some(account_keys) = serde_json::from_str::<serde_json::Value>(account_indexation)?.as_array() {
+            for account_key in account_keys {
+                let account_key = if stronghold {
+                    key_to_chrysalis_key(account_key["key"].as_str().expect("key must be a string").as_bytes())
+                } else {
+                    account_key["key"]
+                        .as_str()
+                        .expect("key must be a string")
+                        .as_bytes()
+                        .to_vec()
+                };
+
+                if let Some(account_data) = chrysalis_data.get(&account_key) {
+                    let account_data = serde_json::from_str::<serde_json::Value>(account_data)?;
+                    if secret_manager_dto.is_none() {
+                        let dto = match &account_data["signerType"]["type"].as_str() {
+                            Some("Stronghold") => format!(
+                                r#"{{"Stronghold": {{"password": null, "timeout": null, "snapshotPath": "{}/wallet.stronghold"}} }}"#,
+                                storage_path.to_string_lossy()
```

Comment:
Should we move it though? Wouldn't it end up inside the database folder itself otherwise?

Reply:
Move it where? Like this it would be the same as in chrysalis

Reply:
Well not exactly. There used to be a /db for the database and the snapshot would be outside. Now it's together with the rocksdb files.

Reply:
The rocksdb folder moved one folder up, but stronghold stays at the exact same place